app/src/main/java/com/lla/panicbutton/ui/fragments/UploadFragment.kt

```diff
@@ -39,24 +58,88 @@ class UploadFragment : Fragment(R.layout.fragment_upload) {
             if (data != null && data.data != null) {
-                val audioFileUri: Uri? = data.data
-                // Now you can use that Uri to get the file path, or upload it, ...
+                //TODO: remove all !!
+                val audioFileUri: Uri = data.data!!
+                val fileName = getFileNameAndSize(audioFileUri).first
+                var size = getFileNameAndSize(audioFileUri).second
+                if (validSize(size)) {
+                    var audioLength = audioFileUri.path?.let { getAudioLengthMillis(it) }
+                    //var uploadedRecording = Recording(fileName, length, Calendar.getInstance().timeInMillis)
+                    Snackbar.make(
+                        this@UploadFragment.requireView(),
+                        "Audio $fileName uploaded.",
+                        Snackbar.LENGTH_LONG
+                    ).show()
+                } else {
+                    Snackbar.make(
+                        this@UploadFragment.requireView(),
+                        "Audio file was too big to upload.",
+                        Snackbar.LENGTH_LONG
+                    ).show()
+                }
+            }
+        }
+    }
+
+    private fun getFileNameAndSize(uri: Uri): Pair<String, Long> {
+        var name = ""
+        var size = 0L
+        if (uri.scheme == "content") {
+            val cursor: Cursor? = context?.contentResolver?.query(uri, null, null, null, null)
+            cursor.use { cursor ->
+                if (cursor != null && cursor.moveToFirst()) {
+                    name = cursor.getString(cursor.getColumnIndex(OpenableColumns.DISPLAY_NAME))
+                    size = cursor.getLong(cursor.getColumnIndex(OpenableColumns.SIZE))
+                }
             }
         }
+        if (name.isEmpty() || size == 0L) {
+            //TODO: should we throw errors or log warnings?
+        }
+        return Pair(name, size)
+    }
+
+    private fun validSize(size: Long): Boolean {
+        //TODO: add max file size!
+        return true
+    }
+
+    private fun getAudioLengthMillis(path: String): Long {
+        //TODO: this doesn't work to get the audio's length
+        val mediaMetadataRetriever = MediaMetadataRetriever()
+        mediaMetadataRetriever.setDataSource(path)
+        return mediaMetadataRetriever.extractMetadata(MediaMetadataRetriever.METADATA_KEY_DURATION)!!
+            .toLong()
+    }
+
+    //TODO: doesn't work
+    private fun createRecordingsDirInStorage(context: Context, folderName: String): File {
+        // we are creating the dir where we'll be storing the user's recordings and returning it to write to it
+        val recordingsDir = File(
+            context.getExternalFilesDir(
+                Environment.DIRECTORY_MUSIC
+            ), folderName
+        )
+        if (!recordingsDir.mkdirs()) {
+            Timber.d("${Constants.WARN_LOG_TAG} Directory not created")
+        }
+        return recordingsDir
```

Comment:
@lisa7012 @Zandrenel could you help me with these two? haven't worked on this in a while, but need a second pair of eyes on these two. getting the length of audio file uploaded and creating a dir in the user's phone to store the audio they uploaded

Reply:
you could directly pull the branch if u want